### DISCUSSION TITLE
When using dataloader only works when num_workers = 0, Getting a very weird errr when not 0

### DISCUSSION THREAD

**Discussion by PoyrazTahan** [mrdbourke/pytorch-deep-learning#422] [Q&A]
Hey everyone,

I am getting a very strange error. When replicating this moment on my vs code
https://youtu.be/V_xro1bcAuA?t=77030

train_dataloader = DataLoader(dataset=train_data,
                              batch_size=BATCH_SIZE,
                              num_workers=0,
                              shuffle=True)

When I do run with 1 as Daniel does, for some reason it runs the code above, calling bunch of different print statements Daniel has given, like calling walk_through_dir function Daniel custom written. and throws an exception due to crash.

Using: torch: '1.13.1' torchvision 0.14.1

Did anyone encountered any similar problem?

![Screenshot 2023-04-25 at 19 35 25](https://user-images.githubusercontent.com/23452166/234370435-0f37b6ac-38fb-41df-aab5-de0868ef673e.png)
![Screenshot 2023-04-25 at 19 35 32](https://user-images.githubusercontent.com/23452166/234370442-5bad210c-eeed-4a5f-a8ae-581b7ca67e8e.png)
![Screenshot 2023-04-25 at 19 35 04](https://user-images.githubusercontent.com/23452166/234370444-029880b8-5643-4f8e-9ccc-c21c2ba28401.png)
![Screenshot 2023-04-25 at 19 36 21](https://user-images.githubusercontent.com/23452166/234370607-9eeaf43e-1503-49cd-bca2-10d0d6bae6c2.png)

**Comment by PoyrazTahan** [comment]
So, I did some googling I have no Fricking idea how this works but if you import: 
`from tqdm import tqdm`
It fixes the multi-process loading issue.

Any ideas?

https://discuss.pytorch.org/t/error-while-multiprocessing-in-dataloader/46845/31
